### ISSUE TITLE
Missing logo file cause CSFR error in debug mode.

### ISSUE THREAD

**Issue by millipedia** [processwire/processwire-issues#1615]
I posted this bug to the forum so you can find more details and a screencast here:

https://processwire.com/talk/topic/27487-missing-logo-file-causes-csfr-error-in-debug-mode/#comment-226138

But essentially, if you set the custom logo file (logoURL) in AdminThemeUikit settings to a non-existant file then trying to log in to the admin area of the site will result in a CSFR error.

Steps to reproduce:

1. Make sure debug mode is on.
2. Set the logoURL field in AdminThemeUikit settings to a custom file which does not currently exist on the server.
3. Logout.
4. Try to log in and get the 'This request appears to be forged' error.
5. Upload the missing file to the server.
6. You can now log in with no error.

Turning debug mode off enables you to log in with no errors so my first guess is that some PHP warning / error message being output is borking some js or something, but I've not found anything like that yet.

* ProcessWire version: 	3.0.203 dev 
* PHP version: 7.3

**Comment by ryancramerdesign** [comment]
@millipedia I can't seem to duplicate the issue here. I tried two different non-existing files (below) but neither interrupted the login process (debug mode is enabled).
~~~~
/site/templates/styles/images/some-missing-file.png
/some/missing/file.jpg
~~~~

**Comment by millipedia** [comment]
@ryancramerdesign I can consistently repeat this using clean installs of 3.0.204 on a Cloudways / Digital Ocean instances running PHP 7.3 and 8.1

One exciting thing I discovered this morning is that the file extension matters: 
Setting the masthead image file to:

`/site/assets/images/logo.png`

will throw an error if logo.png does not exist (as will logo.jpg). But changing the filename to .svg ( /site/assets/images/logo.svg ) will allow you to log in with no issues.

I'll keep investigating when I have time, but I'm happy to give you access to an affected dev site if that would help (although I appreciate you probably have better things to do...).

Thanks for all you efforts though.

s.

**Comment by Toutouwai** [comment]
I tried and also cannot reproduce using Laragon on Windows.

@millipedia, maybe you could test in a different environment because I'm not sure that Cloudways is a good one to use when debugging core issues. I have several sites on Cloudways and had to spend a fair amount of time troubleshooting the environment so that it works with PW. For example, it uses Nginx which is not officially supported by PW and I also had problems with Varnish cache that is on by default and strips cookies from requests to static assets.

I don't remember everything I did but I made these notes in .htaccess relating to Cloudways:

```
# Must disable item 2 (or set static HTML file)
# Must enable item 18 and add css/js extensions
# Must use 9B instead of 9A
# Must use custom rules for 13A
```

**Comment by millipedia** [comment]
You're right - when I tried on a bog standard CPanel environment then I couldn't repeat the problem.

After a bit of poking around with Cloudways I found that disabling Varnish fixed the problem - so thanks for pointing me at that. I have had problems with Varnish before but because I was doing the absolute minimum to try and replicate then I wasn't changing any defaults - that'll teach me.

I'm still curious as to why broken svgs work but not broken png files (I notice there's another issue about error suppression when reading exif data) but that's for another time. I'll close this issue now. 

Thanks for all your help.

s.
